### ISSUE TITLE
CA1508 (Avoid dead conditional code) - not analyzing 'HasValue' of nullable value types

### ISSUE THREAD

**Issue by kindermannhubert** [dotnet/roslyn-analyzers#5003]
### Analyzer

**Diagnostic ID**: [CA1508](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/CA1508): `Avoid dead conditional code`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 5.0.3 (Latest)

### Describe the bug

Missing diagnostic when using 'HasValue' property.
```c#
        static void M()
        {
            int? i = null;

            if (i == null) { } //CA1508: 'i == null' is always 'true'
            if (i != null) { } //CA1508: 'i == null' is always 'false'

            if (!i.HasValue) { } //CA1508 missing
            if (i.HasValue) { }  //CA1508 missing
        }
```

### Expected behavior
Diagnostic is reported even when using 'HasValue'.
